pandas/_libs/period.pyx

```diff
@@ -693,7 +693,12 @@ cdef class _Period(object):
 
     def __richcmp__(self, other, op):
         if is_period_object(other):
-            if other.freq != self.freq:
```

Comment:
why does ``other.freq != self.freq`` *ever* raise?

Reply:
Your guess is as good as mine.  Googling turned up some similar errors in Django, a suggestion that the GIL is involved.  Hopefully we'll get to the root of the problem eventually.  For now I'll be happy to just raise the right error.

Ref #17112, or to reproduce in py3:
```
>>> ser = pd.Series([pd.Period('2017'), pd.Period('2016Q1')])
>>> ser.sort_values()
[...]
SystemError: <built-in function isinstance> returned a result with an error set
```